src/charm.py

```diff
@@ -83,110 +97,113 @@ def __init__(self, *args):
 
-        # Makes sure the directory for the logfile exists
-        if not self._container.isdir(str(LOG_DIR)):
-            self._container.make_dir(path=str(LOG_DIR), make_parents=True)
-            logger.info(f"Created directory {LOG_DIR}")
-
-        self._set_version()
-        self._update_pebble_layer(event)
-
-    def _on_install(self, event: InstallEvent) -> None:
-        if not self._container.can_connect():
-            event.defer()
-            logger.info("Cannot connect to admin-ui container. Deferring the event.")
-            self.unit.status = WaitingStatus("Waiting to connect to admin-ui container")
-            return
-
-        # Makes sure the directory for the logfile exists
-        # Duplicated to avoid race condition between install and pebble_ready events.
-        if not self._container.isdir(LOG_DIR):
-            self._container.make_dir(path=LOG_DIR, make_parents=True)
-            logger.info(f"Created directory {LOG_DIR}")
+        self._handle_status_update_config(event)
 
     def _on_config_changed(self, event: ConfigChangedEvent) -> None:
         """Handle changed configuration."""
-        self._update_pebble_layer(event)
+        self._handle_status_update_config(event)
 
-    def _update_pebble_layer(self, event: HookEvent) -> None:
+    def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+        if self.unit.is_leader():
+            logger.info("This app's public ingress URL: %s", event.url)
+
+    def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
+        if self.unit.is_leader():
+            logger.info("This app no longer has ingress")
+
+    def _handle_status_update_config(self, event: HookEvent) -> None:
         if not self._container.can_connect():
             event.defer()
             logger.info("Cannot connect to admin-ui container. Deferring the event.")
             self.unit.status = WaitingStatus("Waiting to connect to admin-ui container")
             return
 
-        self.unit.status = MaintenanceStatus("Configuration in progress")
+        self.unit.status = MaintenanceStatus("Configuring the container")
+
+        # Make sure the directory for the logfile exists
+        if not self._container.isdir(str(LOG_DIR)):
+            self._container.make_dir(path=str(LOG_DIR), make_parents=True)
+            logger.info(f"Created directory {LOG_DIR}")
 
-        self._container.add_layer(SERVICE_NAME, self._admin_ui_pebble_layer, combine=True)
+        self._container.add_layer(
+            WORKLOAD_CONTAINER_NAME, self._admin_ui_pebble_layer, combine=True
+        )
         logger.info("Pebble plan updated with new configuration, replanning")
+
         try:
             self._container.replan()
         except ChangeError as err:
             logger.error(str(err))
             self.unit.status = BlockedStatus("Failed to replan, please consult the logs")
             return
 
+        self._set_version()
```

Comment:
Perhaps `pebble_ready` is better for this? Are there any cases were that wouldn't work?